curriculum/challenges/english/25-front-end-development/workshop-plant-nursery-catalog/6734f0ae9d1cf4dfd43a7805.md

```diff
@@ -0,0 +1,143 @@
+---
+id: 6734f0ae9d1cf4dfd43a7805
+title: Step 15
+challengeType: 0
+dashedName: step-15
+---
+
+# --description--
+
+To remove an element from a map object, you can use the `delete` method, which takes a key as argument and removes the entry associated with that key. It returns a boolean indicating if the removal has been successful.
+
+Create a function named `removePlant` that takes a plant object as its argument and removes it from the catalog. Make it return either `true` or `false` depending on if the specified plant has been deleted from the catalog.
+
+After that, log the result of calling your new function with the string `"ballerina"` as its argument.
+
+# --hints--
+
+You should have a function named `removePlant`.
+
+```js
+assert.isFunction(removePlant);
+```
+
+Your `removePlant` function should take a single argument.
+
+```js
+assert.lengthOf(removePlant, 1);
+```
+
+Your `removePlant` function should call the `delete` method on `catalog`.
+
+```js
+let flag = false;
+const temp = catalog.delete;
+try {
+    catalog.delete = (p) => {flag = true}
+    removePlant("");
+    assert.isTrue(flag);
+} finally {
+    catalog.delete = temp;
+}
+```
+
+`removePlant(item)` should delete `item` from your catalog.
+
+```js
+const testArr = [];
+const temp = catalog.delete;
+try {
+    catalog.delete = (p) => {testArr.push(p)}
+    removePlant("test");
+    assert.deepEqual(testArr, ["test"]);
+} finally {
+    catalog.delete = temp;
+}
+```
+
+Your `removePlant` function should return a boolean value indicating if the removal has been successful.
+
+```js
+try {
+    assert.isFalse(removePlant("test"));
+    assert.isTrue(removePlant(royalCrown));
+} finally {
+    catalog.set(royalCrown, { small: 40, medium: 22, large: 9 })
```

Comment:
I'm not actually sure what this line of code is intending to do. Aren't tests usually ran in isolation?